site/app/controllers/student/SubmissionController.php

```diff
@@ -840,6 +841,160 @@ public function ajaxDeleteSplitItem($gradeable_id) {
+                    if ($g !== null) {
+                        $team = $g->getSubmitter()->getTeam();
+                        $team_id = $team->getId();
+                        $who_id = $team_id;
+                        $user_id = "";
+                        $is_team = true;
+                    }
+                    else {
+                        return $this->uploadResult("Must be on a team to access submission.", false);
+                    }
+                }
+                if ($regrade_all_students_all === 'true' || $regrade_all === 'true') {
+                    $active_version = $i;
+                }
+                else {
+                    $active_version = $g->getAutoGradedGradeable()->getActiveVersion();
+                }
+                //create file name
+                $queue_file_helper = [$this->core->getConfig()->getSemester(), $this->core->getConfig()->getCourse(),
+                    $gradeable_id, $who_id, $active_version];
+                $queue_file_helper = implode("__", $queue_file_helper);
+                $queue_file = FileUtils::joinPaths(
+                    $this->core->getConfig()->getSubmittyPath(),
+                    "to_be_graded_queue",
+                    $queue_file_helper
+                );
+                //create json file contents
+                $queue_data = [
+                    "course" => $this->core->getConfig()->getCourse(),
+                    "gradeable" => $gradeable_id,
+                    "is_team" => $is_team,
+                    "max_possible_grading_time" => $gradeable->getAutogradingConfig()->getMaxPossibleGradingTime(),
+                    "queue_time" => $this->core->getDateTimeNow()->format("Y-m-d H:i:sO"),
+                    'regrade' => true,
+                    "required_capabilities" => $gradeable->getAutogradingConfig()->getRequiredCapabilities(),
+                    "semester" => $this->core->getConfig()->getSemester(),
+                    "team" => $team_id,
+                    "user" => $user_id,
+                    "vcs_checkout" => false,
+                    "version" => $active_version,
+                    "who" => $who_id
+                ];
+                //add file to directory which will trigger autograding
+                if (@file_put_contents($queue_file, FileUtils::encodeJson($queue_data), LOCK_EX) === false) {
+                    return $this->uploadResult("Failed to create file for grading queue.", false);
+                }
+                //keep track of how many are added to queue
+                $count += 1;
+            }
+        }
+        if ($regrade === 'true' || $regrade_all === 'true') {
+            $msg = $count . " submission(s) from " . $who_id . " added to queue for regrading";
+        }
+        else {
+            $msg = $count . " submissions added to queue for regrading";
+        }
+        return $this->core->getOutput()->renderResultMessage($msg, true);
```

Comment:
This should probably be return JsonResponse::getSuccessResponse

Reply:
This makes it so it displays the nice green success message at the top of the page which Barb wanted.